phabricator.js

```diff
@@ -0,0 +1,157 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+let parentRevision = null;
+let failedToFetch = false;
+
+async function injectButtons() {
+  let blocks = [];
+  let oldCount = 0;
+  do {
+    await wait(1000);
+    oldCount = blocks.length;
+    blocks = document.querySelectorAll('div[data-sigil=differential-changeset]');
+  } while(!blocks.length || oldCount != blocks.length);
```

Comment:
Is this waiting for the page to be fully loaded?

Reply:
Yes, the diffs are loaded dynamically (and somewhat slowly), one at a time, and this waits until there are no more diffs showing up.

Reply:
We could observe modifications on the page with a MutationObserver, so we can react as soon as a diff is loaded.

If that doesn't work, we should reduce the time we wait and we should start loading the data for the blocks early.

Reply:
> If that doesn't work, we should reduce the time we wait and we should start loading the data for the blocks early.

That is, as soon as we detect a new block, we start loading the coverage data for it.

Reply:
Actually, the line file paths and buttons are there when the page loads. What appears later is the file diffs, so we don't need MutationObserver or any waiting mechanisms.